python/cudf/cudf/core/column/column.py

```diff
@@ -2825,8 +2830,12 @@ def as_column(
     if dtype is not None:
         dtype = cudf.dtype(dtype)
         try:
+            # pyarrow cannot infer pd.NA values
+            sanitized_nas = [
+                val if not is_na_like(val) else None for val in arbitrary
+            ]
```

Comment:
```suggestion
```

This would be really inefficient to iterate over the entire data. Can we rather do something else like:

```python
try:
    arbitrary = pa.array(
                arbitrary,
                type=cudf_dtype_to_pa_type(dtype),
                from_pandas=from_pandas,
            )
except (pa.ArrowInvalid, pa.ArrowTypeError) as e:
    if e.startswith("Could not convert <NA> with type NAType"):
        arbitrary = pa.array(arbitrary, type=..., from_pandas=True)
     ...
     existing code.
```

Reply:
Sure thing. Changed it to this.